dace/frontend/python/astutils.py

```diff
@@ -222,10 +223,14 @@ def astrange_to_symrange(astrange, arrays, arrname=None):
                 begin = symbolic.pystr_to_symbolic(0)
             else:
                 begin = symbolic.pystr_to_symbolic(unparse(begin))
+                if isinstance(begin, numbers.Number) and begin < 0:
```

Comment:
(applies to all other checks in the PR) you can check for "definitely under zero" by changing the check to `(begin < 0) == True` (note the `==` rather than `is`)

Reply:
This will also solve the currently-failing cases